### PR TITLE
Adding The Article On Binary Search Questions

### LINKED ISSUE #92: [Feature Request] <Binary Search Pattern Wise Questions>
---

### Problem Description
Binary Search is often considered an easy algorithm to understand but can be quite challenging to apply effectively in real-world scenarios. While the basic concept is simple, the implementation varies widely depending on the problem at hand. I’ve realized that simply knowing the algorithm is not enough; mastering binary search requires familiarity with its various use cases and patterns. To bridge this gap, users need hands-on practice with different binary search problems, each with unique conditions and complexities.

### Proposed Solution
I propose adding a comprehensive set of binary search questions categorized by patterns. Each question will be labeled based on difficulty (easy, medium, hard) and the pattern it follows. This will help users practice and recognize the different ways binary search is used in real-world problems. Additionally, solutions with explanations and links for practice will be provided to help users understand the implementation in depth.

### Alternatives Considered
- Providing only generic binary search problems without categorization, which could make it difficult for users to grasp the range of use cases.
- Focusing solely on theoretical explanations without practical application, which would not help users master the algorithm.

### Additional Context
The feature will also include pattern labels to guide users in understanding which variation of binary search is being used. Links to solutions and practice problems will ensure users have the necessary resources to solidify their knowledge.

---

**Would you like to work on this feature?**  
- [x] Yes  
- [ ] No  

---

### PR DESCRIPTION
**Pull request by PRASHANTSWAROOP001**
### Description

I added a new article that features pattern-wise questions on binary search. It includes links to practice questions and explanation links for better understanding of each question.

Fixes #92

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

### Checklist:

- [ ✔️] My code follows the style guidelines of this project
- [ ✔️] I have performed a self-review of my code
- [ ✔️] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: docs/binary-search/binary_search_questions.md [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+    - [Pattern 2: Binary Search on Answer Space](#pattern-2-binary-search-on-answer-space)
+    - [Pattern 3: Binary Search on 2D Arrays](#pattern-3-binary-search-on-2d-arrays)
+  - [Problem Collections](#problem-collections)
+    - [Pattern 1: Binary Search on 1D Arrays](#pattern-1-binary-search-on-1d-arrays-1)
+    - [Pattern 2: Binary Search on Answer Space](#pattern-2-binary-search-on-answer-space-1)
+    - [Pattern 3: Binary Search on 2D Arrays](#pattern-3-binary-search-on-2d-arrays-1)
+  - [Real-World Applications](#real-world-applications)
+  - [Benefits of Mastering Binary Search](#benefits-of-mastering-binary-search)
+  - [How to Use This Guide](#how-to-use-this-guide)
+
```
Comment (ajay-dhangar) [inline docs/binary-search/binary_search_questions.md]: I'd like to remove the table of contents from this document. We have a default feature for this in the sidebar based on the .md syntax `##` and `###` tags.

File: docs/binary-search/binary_search_questions.md [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+1. Start with the basic binary search implementation in Pattern 1.
+2. Progress through each pattern, solving problems of increasing difficulty.
+3. For each problem:
+   - Attempt to solve it independently.
+   - If stuck, refer to the provided explanation.
+   - After solving, compare your solution with the optimal approach.
+4. Reflect on the patterns and techniques used in each problem.
+5. Try to apply these patterns to new, unseen problems to reinforce your learning.
+
+---
```
Comment (ajay-dhangar) [inline docs/binary-search/binary_search_questions.md]: delete line ``` --- ```

**Review by ajay-dhangar — CHANGES_REQUESTED** [review]
update your code based on suggestions

File: docs/binary-search/binary_search_questions.md (hunk 2 above)
Comment (PRASHANTSWAROOP001) [reply docs/binary-search/binary_search_questions.md]: I will do this ASAP. Thanks For The Feedback.

**Comment by PRASHANTSWAROOP001** [comment]
I have removed the suggested lines from 14 to 31 and the bottom --- line. Thanks For The Review.

**Review by ajay-dhangar — APPROVED** [review]
Done

**Pull request merged**